tsfresh/feature_selection/selection.py

```diff
@@ -140,6 +140,11 @@ def select_features(X, y, test_for_binary_target_binary_feature=defaults.TEST_FO
     if isinstance(y, np.ndarray):
         y = pd.Series(y, index=X.index)
 
+    y = y.sort_index()
```

Comment:
I would move the sorting inside the calculate_relevance_table method. That way it gets more robust